src/ToolUpdate/UpdateExecutor.php

```diff
@@ -154,4 +174,35 @@ private function validateHash(string $pathToPhar, ?ToolHash $hash): void
             throw new RuntimeException('Invalid hash for file: ' . $pathToPhar);
         }
     }
-}
\ No newline at end of file
+
+    private function verifySignature(string $pharPath, ToolInformationInterface $tool): ?string
+    {
+        $signatureUrl = $tool->getSignatureUrl();
+        if (null === $signatureUrl) {
+            // TODO: How should we handle missing signatures:
+            // Show a warning or add a configuration how phpcq should deal with it?
+            return null;
```

Comment:
IMO we should handle missing signatures as fatal unless configured otherwise (eg via whitelist in YAML or the like).

Reply:
Implemented with ea713cc6bbcca5f0a672cd40e10e25561946a2fd.